core/src/test/scala/org/apache/spark/sql/store/UnifiedPartitionerTest.scala

```diff
@@ -0,0 +1,350 @@
+package org.apache.spark.sql.store
+
+import java.math.{BigInteger, BigDecimal}
+import java.sql.DriverManager
+import java.util.Date
+
+import com.gemstone.gemfire.cache.{PartitionResolver, RegionAttributes, Region, CacheFactory, Cache}
+import com.pivotal.gemfirexd.TestUtil
+import com.pivotal.gemfirexd.internal.engine.Misc
+import com.pivotal.gemfirexd.internal.engine.ddl.resolver.GfxdPartitionByExpressionResolver
+import com.pivotal.gemfirexd.internal.iapi.types._
+import io.snappydata.SnappyFunSuite
+import io.snappydata.core.{Data1, TestData2}
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfter}
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.ColumnName
+import org.apache.spark.sql.catalyst.expressions.GenericMutableRow
+import org.apache.spark.sql.execution.QueryExecution
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * This test checks the validity of various functionality when we use Spark's partitioner logic for underlying
+ * GemXD storage.
+ */
+class UnifiedPartitionerTest extends SnappyFunSuite
+with Logging
+with BeforeAndAfter
+with BeforeAndAfterAll {
+
+  after {
+    snc.dropTable(RowTableName2, ifExists = true)
+    snc.dropTable(RowTableName1, ifExists = true)
+    snc.dropTable(ColumnTableName2, ifExists = true)
+    snc.dropTable(ColumnTableName1, ifExists = true)
+
+
+  }
+
+  val ColumnTableName1: String = "ColumnTable1"
+  val ColumnTableName2: String = "ColumnTable2"
+  val RowTableName1: String = "RowTable1"
+  val RowTableName2: String = "RowTable2"
+
+  val props = Map.empty[String, String]
+
+
+  val options = "OPTIONS (PARTITION_BY 'col1')"
+
+  val optionsWithURL = "OPTIONS (PARTITION_BY 'Col1', URL 'jdbc:snappydata:;')"
+
+
+  test(" Test hash codes for all Sql types ") {
+    snc.sql(s"CREATE TABLE $ColumnTableName1(OrderId INT ,ItemId INT, ItemRef INT) " +
+        "USING column " +
+        "options " +
+        "(" +
+        "PARTITION_BY 'OrderId'," +
+        "PERSISTENT 'ASYNCHRONOUS')")
+
+    val regtwo: Region[_, _] = Misc.getRegion("/APP/COLUMNTABLE1", true, true)
+    val rattr: RegionAttributes[_, _] = regtwo.getAttributes
+    val pr: PartitionResolver[_, _] = rattr.getPartitionAttributes.getPartitionResolver
+    val rpr: GfxdPartitionByExpressionResolver = pr.asInstanceOf[GfxdPartitionByExpressionResolver]
+    assert(rpr != null)
+
+    // Check All Datatypes
+    val row = new GenericMutableRow(1)
```

Comment:
Is it row we need to compare against or InternalRow? Exchange partitioning is on InternalRow. See comments about date, timestamp above.